src/sys.rs

```diff
@@ -455,10 +455,39 @@ mod inner {
             call!(SystemTimeToTzSpecificLocalTime(0 as *const _,
                                                   &mut utc, &mut local));
             system_time_to_tm(&local, tm);
-
+            tm.tm_utcoff = 0;
+            tm.tm_isdst = -1;
+            
             let mut tz = mem::zeroed();
-            GetTimeZoneInformation(&mut tz);
-            tm.tm_utcoff = -tz.Bias * 60;
+            let this_year = 1900 + tm.tm_year as u16;
+            let get_tz_info = GetTimeZoneInformationForYear(this_year, 0 as *mut _, &mut tz) != 0;
```

Comment:
Does this normally fail? The general pattern for these functions currently is to panic if they fail (e.g. the `call!` macro)